File: advent-13.py
import webbrowser
from urllib.parse import urlencode

with open("input-13", "r") as f:
    f.readline()
    line = f.readline()
    mods = [
        (int(x), (-i) % int(x)) for (i, x) in enumerate(line.split(",")) if x != "x"
    ]

    # x, y = list(zip(*mods))
    # s = "ChineseRemainder[{{{}}}, {{{}}}]".format(
    #     ",".join(map(str, x)), ",".join(map(str, y))
    # )

    s = ", ".join("Mod[x, {}] == {}".format(*mod) for mod in mods)
    s = "Reduce[{{{}}}, x, Integers]".format(s)
    qstr = urlencode({"i": s})
    webbrowser.open("https://www.wolframalpha.com/input/?{}".format(qstr))

File: advent-13.jl
function chinese_remainder(a::Array{Tuple{Int,Int}})
    p = prod([x[2] for x in a])

    return mod(sum(x * invmod(p ÷ n, n) * p ÷ n for (x, n) in a), p)
end

open("input-13") do file
    start = parse(Int, readline(file))
    buses = [tryparse(Int, x) for x in split(readline(file), ",")]
    real_buses = filter(x -> !isnothing(x), buses)

    min, i = findmin([bus - start % bus for bus in real_buses])
    println(min * real_buses[i])

    println(chinese_remainder([(-i + 1, n) for (i, n) in enumerate(buses) if !isnothing(n)]))
end